src/Lykke.Job.HistoryExportBuilder.Services/CsvMaker.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using CsvHelper;
+using CsvHelper.Configuration;
+using JetBrains.Annotations;
+using Lykke.Job.HistoryExportBuilder.Core.Services;
+using Lykke.Service.Assets.Client;
+using Lykke.Service.OperationsHistory.AutorestClient.Models;
+
+namespace Lykke.Job.HistoryExportBuilder.Services
+{
+    [UsedImplicitly]
+    public class CsvMaker : IFileMaker
+    {
+        private readonly IAssetsServiceWithCache _assetsServiceWithCache;
+
+        public CsvMaker(
+            IAssetsServiceWithCache assetsServiceWithCache)
+        {
+            _assetsServiceWithCache = assetsServiceWithCache;
+        }
+        
+        public async Task<MemoryStream> MakeAsync(IEnumerable<HistoryOperation> operations)
+        {
+            var assets = await _assetsServiceWithCache.GetAllAssetsAsync();
+            var assetPairs = await _assetsServiceWithCache.GetAllAssetPairsAsync();
+            
+            using (var stream = new MemoryStream())
+            {
+                using (var streamWriter = new StreamWriter(stream) {AutoFlush = true})
+                {
+                    var userCsv = new CsvWriter(streamWriter);
+                    userCsv.Configuration.RegisterClassMap<HistoryOperationCsvEntryMap>();
+
+                    userCsv.WriteRecords(operations.Select(x =>
+                    {
+                        var assetPair = assetPairs.FirstOrDefault(y => y.Id == x.AssetPair);
```

Comment:
Convert assetPairs to dictionary - to spead up search and avoid iterating through whole assetPairs collection for each operation